### PR TITLE
Better error message when trying to add a function to a plot.

### LINKED ISSUE #3571: Suggestion for more informative error message when () missing from geom_*
If you omit the `()` at the end of a geom, you get the following error:

``` r
library(ggplot2)

ggplot(mtcars, aes(y = mpg, group = vs)) +
  geom_boxplot
#> Error: Don't know how to add geom_boxplot to a plot
```

I think the error would be more informative if it continued on to say _"Did you mean `geom_boxplot()`?"_

### PR DESCRIPTION
**Pull request by clauswilke**
This closes #3571:

``` r
library(ggplot2)
ggplot() + geom_point
#> Error: Don't know how to add geom_point to a plot. Did you mean geom_point()?
```

<sup>Created on 2019-10-28 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

File: R/plot-construction.r [hunk 1]
```diff
@@ -100,6 +100,13 @@ ggplot_add.data.frame <- function(object, plot, object_name) {
   plot
 }
 #' @export
+ggplot_add.function <- function(object, plot, object_name) {
+  stop(
+    "Don't know how to add ", object_name, " to a plot. Did you mean ",
+    object_name, "()?", call. = FALSE
+  )
```
Comment (yutannihilation) [inline R/plot-construction.r]: Do we use `abort()`? Or will it be done in #3526?

File: NAMESPACE [hunk 2]
```diff
@@ -105,6 +106,7 @@ S3method(makeContent,labelgrob)
 S3method(makeContext,dotstackGrob)
 S3method(merge_element,default)
 S3method(merge_element,element)
+S3method(merge_element,element_blank)
```
Comment (yutannihilation) [inline NAMESPACE]: These diff seems to come from #3570, but I'm OK whether or not they are included in this PR.

File: R/plot-construction.r (hunk 1 above)
Comment (clauswilke) [reply R/plot-construction.r]: As of now, we're using `stop()`. See e.g. here (purposefully not using permalink):
https://github.com/tidyverse/ggplot2/blob/master/R/plot-construction.r#L89-L92

File: NAMESPACE (hunk 2 above)
Comment (clauswilke) [reply NAMESPACE]: Good catch. I opened a separate PR to disentangle these, see #3595.

File: R/plot-construction.r (hunk 1 above)
Comment (yutannihilation) [reply R/plot-construction.r]: OK.

**Pull request merged**